statistics/handle.go

```diff
@@ -309,3 +317,85 @@ func (h *Handle) FlushStats() {
 		logutil.Logger(context.Background()).Debug("[stats] dump stats feedback fail", zap.Error(err))
 	}
 }
+
+// AnalyzeJob is used to represent the status of one analyze job.
+type AnalyzeJob struct {
+	sync.Mutex
+	DBName        string
+	TableName     string
+	PartitionName string
+	JobInfo       string
+	RowCount      int64
+	StartTime     time.Time
+	State         string
+}
+
+const (
+	pending = "pending"
+	running = "running"
+	succeed = "succeed"
+	failed  = "failed"
+)
+
+// AddNewAnalyzeJob adds new analyze job.
+func (h *Handle) AddNewAnalyzeJob(job *AnalyzeJob) {
+	job.State = pending
+	h.analyzeJobs.Lock()
+	defer h.analyzeJobs.Unlock()
+	h.analyzeJobs.jobs[job] = struct{}{}
+}
+
+// MoveToHistory moves the analyze job to history.
+func (h *Handle) MoveToHistory(job *AnalyzeJob) {
+	h.analyzeJobs.Lock()
+	delete(h.analyzeJobs.jobs, job)
+	h.analyzeJobs.history[job] = struct{}{}
+	h.analyzeJobs.Unlock()
+}
+
+// ClearHistoryJobs clears all history jobs.
+func (h *Handle) ClearHistoryJobs() {
+	h.analyzeJobs.Lock()
+	h.analyzeJobs.history = make(map[*AnalyzeJob]struct{})
+	h.analyzeJobs.Unlock()
+}
+
+// GetAllAnalyzeJobs gets all analyze jobs.
+func (h *Handle) GetAllAnalyzeJobs() []*AnalyzeJob {
+	h.analyzeJobs.Lock()
+	defer h.analyzeJobs.Unlock()
+	jobs := make([]*AnalyzeJob, 0, len(h.analyzeJobs.jobs)+len(h.analyzeJobs.history))
+	for job := range h.analyzeJobs.jobs {
+		jobs = append(jobs, job)
+	}
+	for job := range h.analyzeJobs.history {
+		jobs = append(jobs, job)
+	}
+	return jobs
+}
+
+// Start marks status of the analyze job as running and update the start time.
+func (job *AnalyzeJob) Start() {
+	job.Mutex.Lock()
+	job.State = running
```

Comment:
Do we need to check whether the job is `running`? It looks like a job may `Start()` several times.

Reply:
When will it start several times?

Reply:
Oh, never mind. Different tasks have Different jobs.